pkg/kv/dist_sender.go

```diff
@@ -516,6 +516,10 @@ func (ds *DistSender) initAndVerifyBatch(
 		if len(ba.Txn.ObservedTimestamps) == 0 {
 			// Ensure the local NodeID is marked as free from clock offset;
 			// the transaction's timestamp was taken off the local clock.
+			// TODO(andrei): This is broken when Txn.OrigTimestamp has not been, in
```

Comment:
Could you explain why it is broken? What can happen in the case where the `txn.OrigTimestamp` was set from another node's clock?